src/configuration.ts

```diff
@@ -1,24 +1,41 @@
 import { workspace } from 'vscode';
 import * as path from 'path';
 import { existsSync } from 'fs';
+import semver = require('semver');
+import { execSync } from 'child_process';
+import { outputChannel } from './extension';
 
 export interface AmebaConfig {
     command: string;
     configFileName: string;
     onSave: boolean;
+    onType: boolean;
 }
 
 export function getConfig(): AmebaConfig {
     let command = 'ameba';
     const root = workspace.workspaceFolders || [];
     if (root.length) {
         const localAmebaPath = path.join(root[0].uri.fsPath, 'bin', 'ameba');
-        if (existsSync(localAmebaPath)) command = localAmebaPath;
+        if (existsSync(localAmebaPath)) {
+            outputChannel.appendLine(`[Config] Using local ameba at ${localAmebaPath}`)
+            command = localAmebaPath;
+        } else {
+            outputChannel.appendLine(`[Config] Using system ameba`)
+        }
     }
 
+    const workspaceConfig = workspace.getConfiguration('crystal-ameba');
+    const currentVersion = execSync(`"${command}" --version`).toString();
+
+    // Support added in ameba v1.6.3
+    let onType = workspaceConfig.get<boolean>("onType", true) &&
+        semver.satisfies(currentVersion, ">=1.6.3")
+
     return {
         command,
         configFileName: '.ameba.yml',
-        onSave: true
+        onSave: true,
```

Comment:
Shouldn't this value be taken from the settings as well?

Reply:
There wasn't an existing setting for that, but I can add one.

Reply:
I'd do so if there are no downsides to that.

Reply:
The hard part is the `onDidChangeTextDocument` event runs when typing and when saving, so it's not always accurate to determine the real reason for the change. Not supporting the case of `onType == true` and `onSave == false` would make this simpler, and that case doesn't make sense from an end user perspective.